models/iaf_cond_alpha_mc_kinetics.cpp

```diff
@@ -0,0 +1,875 @@
+            ? node.P_.g_conn[ n ] * ( ( V - node.P_.E_L[ n ] )
+                                      - ( y[ S::idx( n + 1, S::V_M ) ] - node.P_.E_L[ n + 1 ] ) )
+            : 0 );
+
+    // derivatives
+    // membrane potential
+    f[ S::idx( n, S::V_M ) ] =
+      ( -I_L - I_syn_exc - I_syn_inh - I_conn + node.B_.I_stim_[ n ] + node.P_.I_e[ n ] + ica
+        + y[ S::idx( n, S::I_AP ) ] ) / node.P_.C_m[ n ];
+
+    // excitatory conductance
+    f[ S::idx( n, S::DG_EXC ) ] = -y[ S::idx( n, S::DG_EXC ) ] / node.P_.tau_synE[ n ];
+    f[ S::idx( n, S::G_EXC ) ] =
+      y[ S::idx( n, S::DG_EXC ) ] - y[ S::idx( n, S::G_EXC ) ] / node.P_.tau_synE[ n ];
+
+    // inhibitory conductance
+    f[ S::idx( n, S::DG_INH ) ] = -y[ S::idx( n, S::DG_INH ) ] / node.P_.tau_synI[ n ];
+    f[ S::idx( n, S::G_INH ) ] =
+      y[ S::idx( n, S::DG_INH ) ] - y[ S::idx( n, S::G_INH ) ] / node.P_.tau_synI[ n ];
+
+    // active current during AP
+    f[ S::idx( n, S::DI_AP ) ] = -y[ S::idx( n, S::DI_AP ) ] / node.P_.tau_curAP[ n ];
+    f[ S::idx( n, S::I_AP ) ] =
+      y[ S::idx( n, S::DI_AP ) ] - y[ S::idx( n, S::I_AP ) ] / node.P_.tau_curAP[ n ];
+
+    if ( n == N::DIST )
+    {
+      if ( node.P_.act_flag == 1.0 )
+      {
+        f[ S::idx( n, S::M_CA ) ] = ( minf - y[ S::idx( n, S::M_CA ) ] ) / node.P_.tau_m;
+        f[ S::idx( n, S::H_CA ) ] = ( hinf - y[ S::idx( n, S::H_CA ) ] ) / node.P_.tau_h;
+      }
+      else
+      {
+        f[ S::idx( n, S::M_CA ) ] = 0.0;
+        f[ S::idx( n, S::H_CA ) ] = 0.0;
+      }
+    }
+    else
+    {
+      f[ S::idx( n, S::M_CA ) ] = 0.0;
+      f[ S::idx( n, S::H_CA ) ] = 0.0;
+    }
+  }
+
+  return GSL_SUCCESS;
+}
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
+
+nest::iaf_cond_alpha_mc_kinetics::Parameters_::Parameters_()
+  : V_th( -55.0 )    // mV
+  , V_reset( -60.0 ) // mV
+  , t_ref( 2.0 )
+  , // ms
+  // changes
+  // max voltage at spike
+  V_max( 30.0 )
+  , // mV
+  E_Ca( 50.0 )
+  , // mV
+  G_Ca( 70.0 )
+  , // nS
+  tau_m( 15.0 )
+  , // ms
+  tau_h( 80.0 )
+  , // ms
+  half_m( -9.0 )
+  , // mV
+  half_h( -21.0 )
+  , // mV
+  slope_m( 0.5 )
+  , // mV-1
+  slope_h( -0.5 )
+  , // mV-1
+  jump_Th( 25.0 )
+  , // mV
+  tau_Th( 7.0 )
+  , // ms
+  act_flag( 1.0 )
+  , reset_flag( 1.0 )
+
+{
+  // conductances between compartments
+  g_conn[ SOMA ] = 2.5; // nS, soma-proximal
+  g_conn[ PROX ] = 1.0; // nS, proximal-distal
+
+  // soma parameters
+  t_L[ SOMA ] = 500.0;    // nS
+  nt_L[ SOMA ] = 15.0;    // nS
+  g_L[ SOMA ] = 10.0;     // nS
+  C_m[ SOMA ] = 150.0;    // pF
+  E_ex[ SOMA ] = 0.0;     // mV
+  E_in[ SOMA ] = -85.0;   // mV
+  E_L[ SOMA ] = -70.0;    // mV
+  tau_synE[ SOMA ] = 0.5; // ms
+  tau_synI[ SOMA ] = 2.0; // ms
+  I_e[ SOMA ] = 0.0;      // pA
+  // changes
+  tau_curAP[ SOMA ] = 1.0; // ms
```

Comment:
This is a current, isn't it? Wouldn't `tau_currAP` then be a better name? Applies also to `amp_curAP` and the corresponding variables in fixedca.